lib/opensusebasetest.pm

```diff
@@ -362,40 +357,27 @@ sub wait_boot {
         # because of broken firmware, bootindex doesn't work on aarch64 bsc#1022064
         push @tags, 'inst-bootmenu' if ((get_var('USBBOOT') and get_var('UEFI')) || (check_var('ARCH', 'aarch64') and get_var('UEFI')) || get_var('OFW'));
         $self->handle_uefi_boot_disk_workaround if (get_var('MACHINE') =~ qr'aarch64' && get_var('BOOT_HDD_IMAGE') && !$in_grub);
-        check_screen(\@tags, $bootloader_time);
+        assert_screen(\@tags, $bootloader_time);
         if (match_has_tag("bootloader-shim-import-prompt")) {
             send_key "down";
             send_key "ret";
-            assert_screen "grub2", 15;
-        }
-        elsif (match_has_tag("migration-source-system-grub2") or match_has_tag('grub2')) {
-            send_key "ret";    # boot to source system
+            assert_screen 'grub2', 15;
         }
         elsif (get_var("LIVETEST")) {
             # prevent if one day booting livesystem is not the first entry of the boot list
             if (!match_has_tag("boot-live-" . get_var("DESKTOP"))) {
                 send_key_until_needlematch("boot-live-" . get_var("DESKTOP"), 'down', 10, 5);
             }
-            send_key "ret";
         }
         elsif (match_has_tag('inst-bootmenu')) {
             # assuming the cursor is on 'installation' by default and 'boot from
             # harddisk' is above
             send_key_until_needlematch 'inst-bootmenu-boot-harddisk', 'up';
             boot_local_disk;
             assert_screen 'grub2', 15;
-            # confirm default choice
-            send_key 'ret';
-        }
-        elsif (match_has_tag('encrypted-disk-password-prompt')) {
```

Comment:
> Makes sure the bootloader appears and then stop in the boot menu.
So why can we remove the handling of encrypted /boot here?

Reply:
I don't know. I guess we should not. I assume this section was added since I created the code originally and this was done wrong by me in the rebase. Probably I need to do the method extraction again by hand.